### PR TITLE
Make burnin detection 'externally' usable

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Refactor the burnin detection example so that the core process can be embedded into other pipelines. This also introduces installing pipelines and model files as shared data.

### CODE REVIEW

**Comment by mwoehlke-kitware** [comment]
I'd also appreciate thoughts from someone with more sprokit knowledge. I did look, briefly, into clusters, but it seems that, besides also not having any existing distribution thereof (as we don't for pipelines, before this), the mechanism to locate them appears inconsistent with the approach we've been working toward, which is for everyone to use the config search paths. Also, it seems we would lose the ability to hook into the cluster's guts, which we have with this approach (as in the commented out bits in the example `.pipe` to write out intermediate images for debugging purposes).

That said, it would be nice if there was a way we could do better in terms of naming. However, identifiers presently must consist of letters, digits, and `_`. (The code suggests that `-` should also be allowed, although I ran into errors when I attempted that.)

**Comment by russelldj** [comment]
Looks good to me. It's useful to see how a well-configured pipeline looks.

That said, the Mac build is failing with the following error:
```
CMake Error at config/CMakeLists.txt:61 (add_subdirectory):
  The source directory

    /Users/kitware/Jenkins/workspace/KwiverMacPR/config/models

  does not contain a CMakeLists.txt file.
```
My CMake knowledge is limited but I was surprised that it seemed rather challenging to simply install a folder of data. Oddly, this works on Linux.

**Comment by mwoehlke-kitware** [comment]
> the Mac build is failing

Oops, forgot to `git add` the `CMakeLists.txt` :flushed:. Should be fixed now.

> Oddly, this works on Linux.

That is indeed very odd. AFAIK, it would fail on *my* (Fedora) system.

File: config/pipelines/detect_burnin.pipe [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# =============================================================================
+process detect_burnin__in
```
Comment (mleotta) [inline config/pipelines/detect_burnin.pipe:2]: Is this double underscore intentional?

Comment (mleotta) [reply config/pipelines/detect_burnin.pipe:2]: It seems to be that is the pattern here.  Is this the normal naming convention for sprokit processes?

Comment (mwoehlke-kitware) [reply config/pipelines/detect_burnin.pipe:2]: Please see (and consider replying to) my first comment.

Process names need to be unique. Thus, simply naming the internal processes e.g. "filter" seems collision prone. What we need are namespaced processes, but we have a very limited character set available for naming. Thus, I went with `__` as the "namespace separator". If you have a better idea, however...

I don't know that this pattern (designing a `.pipe` to be `include`d) has been used before...

**Comment by mleotta** [comment]
I'm not sure why this was only found on the Windows build, but there is a CMake configuration error because we moved the model file from examples but didn't remove the CMake entry in examples for it:

https://github.com/Kitware/kwiver/blob/eed9a3559be7f2352dd093d2694e2fe9770f973b/examples/CMakeLists.txt#L32

**Comment by mleotta** [comment]
> I'd also appreciate thoughts from someone with more sprokit knowledge. I did look, briefly, into clusters, but it seems that, besides also not having any existing distribution thereof (as we don't for pipelines, before this), the mechanism to locate them appears inconsistent with the approach we've been working toward, which is for everyone to use the config search paths. Also, it seems we would lose the ability to hook into the cluster's guts, which we have with this approach (as in the commented out bits in the example `.pipe` to write out intermediate images for debugging purposes).
> 
> That said, it would be nice if there was a way we could do better in terms of naming. However, identifiers presently must consist of letters, digits, and `_`. (The code suggests that `-` should also be allowed, although I ran into errors when I attempted that.)

I think it is beyond the scope of this branch, but we should revisit the concept of clusters and either make them usable and start using them or deprecate them in favor of just including pipe files and then improve process namespaces.  This is a good topic for the KWIVER monthly meeting.  Clusters were originally intended for cases like this, but if no one has tried to use them that might be good evidence that they are not needed.  I don't remember all the details, but I think clusters were supposed to be more powerful than just including pipe files in that you could build a cluster in code and register it and then others could introspect and use it like it was a process.  However, I don't know how common it is to build pipeline in code rather than parsing a `.pipe` file.  There is certainly a lot of overlap between the ideas of clusters and includable `.pipe` files.

**Comment by mwoehlke-kitware** [comment]
> I'm not sure why this was only found on the Windows build

Odd. Should be fixed; thanks!

File: config/pipelines/detect_burnin.pipe (hunk 1 above)
Comment (mleotta) [reply config/pipelines/detect_burnin.pipe:2]: Thanks,  I understand now.  This is fine.  We may want to formalize this convention in docs at some point later, if it sticks.

**Review by mleotta — APPROVED** [review]
If this looks okay to @russelldj then I think it's good to merge.

**Review by russelldj — APPROVED** [review]

**Pull request merged**